packages/grpc-js/src/resolver-ip.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isIPv4, isIPv6 } from "net";
+import { StatusObject } from "./call-stream";
+import { ChannelOptions } from "./channel-options";
+import { LogVerbosity, Status } from "./constants";
+import { Metadata } from "./metadata";
+import { registerResolver, Resolver, ResolverListener } from "./resolver";
+import { SubchannelAddress } from "./subchannel";
+import { GrpcUri, splitHostPort, uriToString } from "./uri-parser";
+import * as logging from './logging';
+
+const TRACER_NAME = 'ip_resolver';
+
+function trace(text: string): void {
+  logging.trace(LogVerbosity.DEBUG, TRACER_NAME, text);
+}
+
+const IPV4_SCHEME = 'ipv4';
```

Comment:
Sorts of feels weird that these are private strings here, but I guess there might not be any needs to expose a list of all the known schemes somewhere.

Reply:
If they were going to be exposed in any way, it would probably be through the `registerResolver` function, so the string constants in this file would still be private.